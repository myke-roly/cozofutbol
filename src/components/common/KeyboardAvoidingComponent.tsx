import React, { useState } from 'react'
import {
  View,
  KeyboardAvoidingView,
  Dimensions,
  Platform,
  KeyboardAvoidingViewProps,
  StyleSheet,
} from 'react-native'

const windowHeight = Dimensions.get('window').height

// TODO: CREATE DINAMIC SCROLL VIEW TO SET HEIGHT ON KEYBOARD OPEN AND CLOSE

const KeyboardAvoidingComponent = ({
  keyboardVerticalOffset = 0,
  children,
  ...props
}: KeyboardAvoidingViewProps) => {
  const [verticalOffset, setVerticalOffset] = useState<number | undefined>(
    keyboardVerticalOffset,
  )

  console.log('render KeyboardAvoidingComponent', verticalOffset)

  return (
    <View
      style={styles.container}
      testID="keyboard-avoiding-view"
      onLayout={({ nativeEvent: { layout } }) =>
        setVerticalOffset(windowHeight - layout.height)
      }>
      <KeyboardAvoidingView
        style={styles.container}
        behavior={Platform.select({ ios: 'padding' })}
        keyboardVerticalOffset={verticalOffset}
        {...props}>
        {children}
      </KeyboardAvoidingView>
    </View>
  )
}

const styles = StyleSheet.create({
  container: { flex: 1 },
})

export default KeyboardAvoidingComponent
