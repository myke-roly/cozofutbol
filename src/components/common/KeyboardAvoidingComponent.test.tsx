import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'
import { View } from 'react-native'
import KeyboardAvoidingComponent from './KeyboardAvoidingComponent'
import { Text } from './'

describe('KeyboardAvoidingComponent', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <KeyboardAvoidingComponent>
          <View>
            <Text>Children custom keyboard avoiding</Text>
          </View>
        </KeyboardAvoidingComponent>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should render correctly sets height onLayout', () => {
    const nativeEventMock = {
      nativeEvent: {
        layout: {
          height: 100,
        },
      },
    }

    const { toJSON, getByTestId } = render(
      <KeyboardAvoidingComponent>
        <View>
          <Text>Children custom keyboard avoiding</Text>
        </View>
      </KeyboardAvoidingComponent>,
    )

    fireEvent(
      getByTestId('keyboard-avoiding-view'),
      'onLayout',
      nativeEventMock,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
