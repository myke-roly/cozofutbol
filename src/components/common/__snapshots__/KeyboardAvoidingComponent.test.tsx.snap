// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardAvoidingComponent Should render correctly sets height onLayout 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flex": 1,
    }
  }
  testID="keyboard-avoiding-view"
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View>
      <Text
        style={
          [
            undefined,
            {
              "fontSize": 16,
            },
          ]
        }
      >
        Children custom keyboard avoiding
      </Text>
    </View>
  </View>
</View>
`;

exports[`KeyboardAvoidingComponent should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flex": 1,
    }
  }
  testID="keyboard-avoiding-view"
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View>
      <Text
        style={
          [
            undefined,
            {
              "fontSize": 16,
            },
          ]
        }
      >
        Children custom keyboard avoiding
      </Text>
    </View>
  </View>
</View>
`;
